draft-ietf-quic-transport.md

```diff
@@ -943,47 +940,104 @@ The fields in the ACK frame are as follows:
 * Ack Delay: Time from when the largest acked, as indicated in the Largest Acked
   field, was received by this peer to when this ack was sent.
 
-* Ack Block Section:
+* Ack Block Section: Contains one or more blocks of packet numbers which have
+  been successfully received.  See {{ack-block-section}}.
+
+* Timestamp Section: Contains zero or more timestamps reporting transit delay of
+  received packets.  See {{timestamp-section}}.
+
 
-  * Num Blocks (opt): An optional 8-bit unsigned value specifying the number of
-    additional ack blocks (besides the required First Ack Block) in this ACK
-    frame.  Only present if the 'N' flag bit is 1.
+### Ack Block Section {#ack-block-section}
 
-  * First Ack Block Length: An unsigned packet number delta that indicates the
-    number of contiguous additional packets being acked starting at the Largest
-    Acked.
+The Ack Block Section contains between one and 256 blocks of packet numbers
+which have been successfully received.  If the `N` bit in the Type field is 0,
+the Num Blocks field is absent and only the First Ack Block length is present.
+If the `N` bit is 1, the Num Blocks field indicates how many additional blocks
+follow the First Ack Block Length field.
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|[Num Blocks(8)]|      First Ack Block Length (8/16/32/48)    ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|  [Gap 1 (8)]  |       [Ack Block 1 Length (8/16/32/48)]     ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|  [Gap 2 (8)]  |       [Ack Block 2 Length (8/16/32/48)]     ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+                             ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|  [Gap N (8)]  |       [Ack Block N Length (8/16/32/48)]     ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
```

Comment:
@janaiyengar, I think your comment is for a different change. My other note said you'd file an issue for reducing the size of the `Time Since Largest Acked` field to match the others, and I agree that should be saved for a different PR.  This is a purely editorial change which you suggested, putting the counts up in the top section.  If you no longer want that, we can revert it.